src/hardware/mixer.cpp

```diff
@@ -1983,8 +1983,11 @@ AudioFrame MixerChannel::Sleeper::MaybeFadeOrListen(const AudioFrame& frame)
 	if (!had_signal) {
 		// Otherwise, we simply passively listen for signal
 		constexpr auto SilenceThreshold = 1.0f;
-		had_signal = fabsf(frame.left)  > SilenceThreshold ||
-		             fabsf(frame.right) > SilenceThreshold;
+
+		had_signal = fabsf(frame.left - last_frame.left) > SilenceThreshold ||
+		             fabsf(frame.right - last_frame.right) > SilenceThreshold;
+
```

Comment:
Very nice commit message 👍🏻 

Basically, you've extended this so it treats DC as silence. That's fine, but I will introduce a per-channel averaging DC remover which will obviate the need for this (but it will do no harm).

`SilenceThreshold` reads a bit weird in this altered context; you might wanna change it.

> This also means that these channels will no longer pollute the downstream filters (which was theoretically possible).

Not sure what you means by "pollute". The per-channel LPF/HPF filters don't care about DC, and the master 20 Hz HPF will remove DC completely from the final output. Because we're using floats, DC doesn't matter in the sense that it cannot cause clipping, then we remove it anyway with the master HPF.

The per-channel averaging DC remover will be needed however to remove  "zipper noise" in certain interesting DC-infested scenarios. See #3486 

Anyway, you can leave this in if you want for now.

Reply:
> Basically, you've extended this so it treats DC as silence.

Oh yes. I'll rename the constant to use better terminology (maybe `DeltaThreshold` or `ChangeThreshold`.

> I will introduce a per-channel averaging DC remover which will obviate the need for this (but it will do no harm).

Excellent; right - if a constant DC shift in a signal is removed, so all signals ultimately oscillate around the 0-line, then inspecting absolute values (instead of difference) will again work fine for sleeping.

I noticed with the DAC, some games an demos leave the sound blaster running in DAC mode (which is fine), however the new approach of "sampling the register" means we just keep reading the last written reg value over and over - so it leads to a constant DC until the game or demo decides to write new stuff.  This at least turns off the channel after half a second or so.

> Not sure what you means by "pollute". The per-channel LPF/HPF filters don't care about DC, and the master 20 Hz HPF will remove DC completely from the final output. Because we're using floats, DC doesn't matter in the sense that it cannot cause clipping, then we remove it anyway with the master HPF.

Even better; thanks for that info. Yes - even with this change, I didn't hear any differences, but I wanted to let the channel sleep because it relaxes the CPU usage.

Reply:
Fixed.